truenas_api_client/__init__.py

```
@@ -785,10 +785,13 @@ def wait(self, c: Call, *, callback: _JobCallback | None=None, job: Literal['RET
                     raise c.error
 
             if job:
-                jobobj = Job(self, c.result, callback=callback)
-                if job == 'RETURN':
-                    return jobobj
-                return jobobj.result()
+                if c.result in self._jobs:
```

This might be a race condition. How can we know that we've already received and handled a `core.get_jobs` `ADDED` event by the time we receive method return value?

---

Should we raise an exception in `Job.__init__` instead?

---

With current architecture, there is no way to synchronize these events.

We also plan to get rid of jobs in 25.10

---

Feel free to close this PR if you determine that it wouldn't be helpful.